experiments/standalone/Soil/evaporation.jl

```diff
@@ -211,60 +214,62 @@ for (FT, tf) in ((Float32, 2 * dt), (Float64, tf))
             S_l_sfc = ClimaLSM.Domains.top_center_to_surface(
                 effective_saturation.(ν, u.soil.ϑ_l, θ_r),
             )
-            layer_thickness =
-                parent(Soil.dry_soil_layer_thickness.(S_l_sfc, S_c, d_ds))[1]
-            push!(dsl, layer_thickness)
+            layer_thickness = Soil.dry_soil_layer_thickness.(S_l_sfc, S_c, d_ds)
+            push!(dsl, parent(layer_thickness)[1])
             push!(r_soil, parent(@. layer_thickness / (_D_vapor * τ_a))[1])
             T_sfc = T_soil[i]
-            ts_in =
-                ClimaLSM.construct_atmos_ts(top_bc.atmos, time, thermo_params)
-            push!(ρ_atmos, Thermodynamics.air_density(thermo_params, ts_in))
-            ρ_sfc = compute_ρ_sfc(thermo_params, ts_in, T_sfc)
-            push!(ρ_soil, ρ_sfc)
-            q_sat = Thermodynamics.q_vap_saturation_generic(
-                thermo_params,
-                T_sfc,
-                ρ_sfc,
-                Thermodynamics.Liquid(),
-            )
-            q_sfc = parent(
-                ClimaLSM.surface_specific_humidity(soil, Y, p, T_sfc, ρ_sfc),
-            )[1]
-            push!(q_soil, q_sfc)
-            ts_sfc = Thermodynamics.PhaseEquil_ρTq(
-                thermo_params,
-                ρ_sfc,
-                T_sfc,
-                q_sfc,
+            ts_in = ClimaLSM.construct_atmos_ts(top_bc.atmos, p, thermo_params)
+            push!(
+                ρ_atmos,
+                parent(Thermodynamics.air_density.(thermo_params, ts_in))[1],
             )
+            ρ_sfc = compute_ρ_sfc.(thermo_params, ts_in, T_sfc)
+            push!(ρ_soil, parent(ρ_sfc)[1])
+            q_sat =
+                Thermodynamics.q_vap_saturation_generic.(
+                    thermo_params,
+                    T_sfc,
+                    ρ_sfc,
+                    Thermodynamics.Liquid(),
+                )
+            q_sfc = ClimaLSM.surface_specific_humidity(soil, Y, p, T_sfc, ρ_sfc)
+
+            push!(q_soil, parent(q_sfc)[1])
+            ts_sfc =
+                Thermodynamics.PhaseEquil_ρTq.(
+                    thermo_params,
+                    ρ_sfc,
+                    T_sfc,
+                    q_sfc,
+                )
             state_sfc =
-                SurfaceFluxes.StateValues(0.0, SVector{2, FT}(0, 0), ts_sfc)
-            state_in = SurfaceFluxes.StateValues(
-                h_atmos,
-                SVector{2, FT}(u_atmos(time), 0),
-                ts_in,
-            )
+                SurfaceFluxes.StateValues.(
+                    0.0,
+                    Ref(SVector{2, FT}(0, 0)),
```

Comment:
It ends up being a Ref because ts_in is now a field, and we need effectively a field of SVectors. Everything is now fields because the atmos state is now a field (in p.drivers) rather than scalar as before.

Reply:
why can't we just pass in the field, as opposed to a Ref of it?

Reply:
hmm so we are calling StateValues.(scalar, svector, field). We want to broadcast over the field, so we need a Ref of the SVector, right? otherwise it doesnt know whether to broadcast over the elements of the field or the elements of the SVector. We want a copy of the Svector at each location in the field.